### PR TITLE
fix: cjs type definition

### PR DESCRIPTION
**Pull request by trim21**
close https://github.com/minio/minio-js/issues/1166

This PR doesn't affect any other PR, can be merged separately.

Also no need to run e2e tests, it only change build script, make it rewrite import path in `.d.ts` files

### CODE REVIEW

**Comment by trim21** [comment]
@prakashsvmx do you mind review this first?

File: package.json [hunk 1]
```diff
@@ -4,7 +4,6 @@
   "description": "S3 Compatible Cloud Storage client",
   "main": "./dist/main/minio.js",
   "module": "./dist/esm/minio.mjs",
-  "types": "./dist/main/minio.d.ts",
```
Comment (aldy505) [inline package.json:7]: I don't think deleting this one would solve the Typescript typings reference issues

Comment (trim21) [reply package.json:7]: we have `./dist/esm/minio.d.mts` for `./dist/esm/minio.mjs` and `./dist/main/minio.d.ts` for `./dist/main/minio.js`, TypeScript will resolve to these types.

Comment (trim21) [reply package.json:7]: I tested this locally and all `moduleResolution` works fine.

**Review by prakashsvmx — APPROVED** [review]

**Comment by trim21** [comment]
consider to create a release for this PR to fix #1163 and https://github.com/minio/minio-js/issues/1166

**Review by kaankabalak — APPROVED** [review]

**Comment by oktapodia** [comment]
@prakashsvmx Any possibility to release this anytime soon?

**Pull request merged**